alteration model change + dashborad + bioMarker prediction script

JiaoJiao123 opened the pull request.

JiaoJiao123 committed:
plotly graph on the dashboard

added 3 graph on the home page to gave a overview about the clinical
trials

JiaoJiao123 committed:
Imported tumor type data and

1: imported tumor category data from the clinical trial gov page. and
display it as bar chart in the home page
2: only display first 3 curation history and expand to show all
3: capitalize each word in recruiting status
4: blue the links

JiaoJiao123 committed:
commit before alteration section change

JiaoJiao123 committed:
change alteration data models (don't merge)

This commit is made because I want to work in another computer and get
the most updated code version. There are still some bugs need to be
fixed. But the changes below are already made.
1: change alteration to inclusion and exclusion
2: save alteration information in the mapping table instead of saving
alterationID

JiaoJiao123 committed:
split alteration to inclusion and exclusion types

1: bio-marker prediction script changed to scan inclusion and exclusion
alterations seperately.
2: directly save gene alteration pair in the mapping table, which
avoided access alteration table too often. If other info about a
alteration needed in the future, we can always added alterationID back.
3: four charts are added in the home page as dash board.
    1) the first bar chart is only for trials with recruiting or
Active, not  recruiting status
     2) the second bar chart is based on the clinical trial cancer type
vocabulary. need to wait for the mapping between oncoKB vocabulary and
clinical trial vocabulary. then it can be converted to the oncoKB
cancer type vocabulary

JiaoJiao123 committed:
Merge remote-tracking branch 'cBioPortal/master' into workBranch

# Conflicts:
#	config/env/all.js

JiaoJiao123 committed:
Merge branch 'JiaoJiao123-workBranch'

JiaoJiao123 committed:
Merge branch 'master' into workBranch

zhx828 commented on .gitignore:
```diff
@@ -7,3 +7,10 @@ public/lib
+
+*.zip
+
+config/env/development.js
```
Why are these files ignored?

zhx828 commented on app/controllers/cancertypes.server.controller.js:
```diff
@@ -156,3 +156,26 @@ exports.hasAuthorization = function(req, res, next) {
 	next();
 };
+function compare(a, b) {
+	if (a.counts < b.counts)
```
Could be ' return b.counts - a.counts;'

zhx828 commented on public/modules/core/views/home.client.view.html:
```diff
@@ -90,14 +93,74 @@
+		<br/>
+	</div>
+
+	<div class="panel panel-default" style="width: 45%; float:left;margin:10px">
```
This style has been repeated several times, you could put them into css file and give class name instead.

zhx828 commented:
I briefly went through the code. I can't test the all of functions at this moment. If possible in the future, could you send more pull requests instead of one? In this case, I can do fully test about each feature you add. I gave some very minor suggestions. Overall, the code looks good to me.

JiaoJiao123 committed:
code formatting

1: move random css style to css file
2: removed some old code
3: find a bug in gene count

JiaoJiao123 committed:
Merge remote-tracking branch 'origin/workBranch' into workBranch

JiaoJiao123 committed:
curator view and assign rules

not done yet. but realize env/development.js is missing. so need to
make commit before pull

JiaoJiao123 committed:
assign rules feature Part 1

This feature is developed to give the curators ways to change the
prediction rules.
1: put each rule in a single file. prediction script read from those
files instead of hard coded.
2: The operations curators made about the rules will change those files.

Notes: Only the interface about this feature is done. The functions to
change files are not realized yet since not sure about how useful this
feature would be.

JiaoJiao123 committed:
assign rule feature completed

This feature will allow curator to assign rules from the front end. The
operations curators made will change the geneAlias.txt and
skipItems.txt which will affect the bio marker prediction accuracy.

JiaoJiao123 committed:
pull trials list API

1: trials list by gene, alteration
2: trials list by gene, alteration and tumor type

JiaoJiao123 committed:
remove irreverent submenus
